packages/orchestration/src/facade.js

```diff
@@ -76,21 +76,19 @@ export const makeOrchestrationFacade = ({
    */
   const orchestrate = (durableName, hostCtx, guestFn) => {
     const subZone = zone.subZone(durableName);
-
-    const hostOrc = makeOrchestrator();
-
-    const [wrappedOrc, wrappedCtx] = prepareEndowment(subZone, 'endowments', [
-      hostOrc,
-      hostCtx,
-    ]);
-
+    const [wrappedCtx] = prepareEndowment(subZone, 'endowments', [hostCtx]);
     const hostFn = asyncFlow(subZone, 'asyncFlow', guestFn);
 
     // cast because return could be arbitrary subtype
     const orcFn = /** @type {HostForGuest<GF>} */ (
-      (...args) => hostFn(wrappedOrc, wrappedCtx, ...args)
+      (...args) => {
+        // each invocation gets a new orchestrator
+        const hostOrc = makeOrchestrator();
+        // TODO: why are the types showing the guest types for arguments?
+        // @ts-expect-error XXX fix broken types
```

Comment:
@turadg you mentioned this was known. Is there an issue we can reference?

Reply:
just a branch https://github.com/Agoric/agoric-sdk/tree/ta/HostOf